### PR TITLE
process: Unshare UTS namespace, and set hostname to 'localhost'

### PR DESCRIPTION
**Pull request by mgorny**
Use UTS namespace to override hostname when network-sandbox is enabled.
Set it to 'localhost' as that has a better chance of being present
in /etc/hosts.  This fixes tests in some packages that try to connect
to localhost via hostname obtained using gethostname(), e.g. docker-py,
and suffer resolution problems due to the system hostname not being
defined in /etc/hosts.

### CODE REVIEW

**Comment by mgorny** [comment]
I'll send it to the ml once travis finishes.

File: lib/portage/process.py [hunk 1]
```diff
@@ -704,6 +706,8 @@ def _exec(binary, mycommand, opt_name, fd_pipes,
 									noiselevel=-1)
 								os._exit(1)
 						if unshare_net:
+							# use 'localhost' to avoid hostname resolution problems
+							socket.sethostname('localhost')
```
Comment (zmedico) [inline lib/portage/process.py]: Maybe we should treat an exception as a warning here.

Comment (zmedico) [reply lib/portage/process.py]: And if it fails, we can include in the message that it's for FEATURES=network-sandbox (like the other `"Unable to unshare: %s (for FEATURES=\"%s\")\n"` message.

File: lib/portage/process.py [hunk 2]
```diff
@@ -704,6 +706,13 @@ def _exec(binary, mycommand, opt_name, fd_pipes,
 									noiselevel=-1)
 								os._exit(1)
 						if unshare_net:
+							# use 'localhost' to avoid hostname resolution problems
+							try:
+								socket.sethostname('localhost')
+							except:
```
Comment (zmedico) [inline lib/portage/process.py]: We better not catch BaseException, so lets `except Exception:`.

File: lib/portage/process.py [hunk 3]
```diff
@@ -704,6 +706,13 @@ def _exec(binary, mycommand, opt_name, fd_pipes,
 									noiselevel=-1)
 								os._exit(1)
 						if unshare_net:
+							# use 'localhost' to avoid hostname resolution problems
+							try:
+								socket.sethostname('localhost')
+							except Exception:
+								writemsg("Unable to set hostname: %s (for FEATURES=\"network-sandbox\")\n" % (
+									errno.errorcode.get(errno_value, '?'),),
```
Comment (zmedico) [inline lib/portage/process.py]: Oh, we want the exception here rather than the errno thing.

**Pull request closed without merging**